### PR TITLE
Update to json data format, add filesystem

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

**Comment by samdark** [comment]
Merge https://github.com/yiisoft/var-dumper/pull/6 first.

File: src/Storage/StorageInterface.php [hunk 1]
```diff
@@ -17,6 +17,8 @@ public function addCollector(CollectorInterface $collector): void;
 
     public function getData(): array;
 
+    public function setDebugId(string $id): void;
```
Comment (xepozz) [inline src/Storage/StorageInterface.php]: I think it's the wrong way.

File: src/Storage/FileStorage.php [hunk 2]
```diff
@@ -4,18 +4,32 @@
...
      * @var CollectorInterface[]
      */
     private array $collectors = [];
+
     private string $path;
 
-    public function __construct(string $path)
+    private ?string $debugId = null;
+
+    private $filesystem;
```
Comment (xepozz) [inline src/Storage/FileStorage.php]: Should be typed

File: src/Storage/FileStorage.php [hunk 3]
```diff
@@ -4,18 +4,32 @@
...
+
+    public function __construct(string $path, FilesystemInterface $filesystem)
     {
         $this->path = $path;
+        $this->filesystem = $filesystem;
+    }
+
+    public function setDebugId(string $id): void
+    {
+        if ($this->debugId === null) {
```
Comment (xepozz) [inline src/Storage/FileStorage.php]: Looks too bad :(

Comment (yiiliveext) [reply src/Storage/FileStorage.php]: Do you have a better one?

File: src/Storage/FileStorage.php (hunk 2 above)
Comment (yiiliveext) [reply src/Storage/FileStorage.php]: Done

File: src/Storage/StorageInterface.php (hunk 1 above)
Comment (yiiliveext) [reply src/Storage/StorageInterface.php]: Do you have a better one?

**Pull request merged**